packages/@aws-cdk/aws-lambda/lib/event-source-mapping.ts

```diff
@@ -332,6 +342,27 @@ export class EventSourceMapping extends cdk.Resource implements IEventSourceMapp
       selfManagedEventSource = { endpoints: { kafkaBootstrapServers: props.kafkaBootstrapServers } };
     }
 
+    let amazonManagedKafkaEventSourceConfig : CfnEventSourceMapping['amazonManagedKafkaEventSourceConfig'];
```

Comment:
Actually, aside from the spacing issue, this doesn't align with the style in this package.

Reply:
What do you mean with style? The type reference to the CfnRessource? Or that I even type these attributes? 
I tryed to do it in the same manner as the kafkaBootstrapServers. 

I now refactored the code a little bit to be less branched. 
Now there are two if statements. Since line 288 makes the properties disjunct only of the paths are possible.

Reply:
I had another look at it and came up with another refactoring that might align better with the overall style. Now the consumerGroupId is validated at the beginning (like the other props) and then set to the CfnRessource at the end. So there is no intermediate variable declaration anymore, but one is still types safe because the CfnRessource complains if the provided object is malformed.